relayer/src/foreign_client.rs

```diff
@@ -875,6 +941,26 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             trusted_height
         };
 
+        // Check if the consensus state at `trusted_height` is within trusting period
+        if let ConsensusStateTrusted::NotTrusted {
+            elapsed,
+            consensus_state_timestmap,
+            network_timestamp,
+        } = self.check_consensus_state_trusting_period(&client_state, &trusted_height)?
+        {
+            warn!(
+                trusted_height = %trusted_height,
+                network_timestmap = %network_timestamp,
+                consensus_state_timestamp = %consensus_state_timestmap,
+                elapsed = ?elapsed,
+                "[{}] cannot build client update message because the provided trusted height is outside of trusting period!",
+                self);
```

Comment:
A bit of inconsistency here. If `trusted_height == client_state.latest_height()` we catch the consensus state expiration above in `self.validated_client_state()` but there is no `warn!` there i think. But if  `trusted_height != client_state.latest_height()` we issue the warning here.
Also why is this not `error!`?

Reply:
Good catch!

ac3dbc09